pkg/deployment/k8s/deployment.go

```diff
@@ -0,0 +1,227 @@
+	endpointSpec := &corev1.Endpoints{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: d.namespace,
+		},
+		Subsets: []corev1.EndpointSubset{
+			{
+				Addresses: []corev1.EndpointAddress{
+					{
+						IP: targetIP, // Replace with the desired IP address of the endpoint.
+					},
+				},
+				Ports: []corev1.EndpointPort{
+					{
+						Port: int32(targetPort),
+					},
+				},
+			},
+		},
+	}
+
+	err := d.client.Create(context.Background(), endpointSpec)
+	if err != nil {
+		d.logger.Errorf("error occurred while creating K8s endpoint %s:", err)
+		return err
+	}
+
+	d.logger.Infof("Creating K8s endPoint at %s:%d that connected to external IP: %s:%d", name, targetPort, targetIP, targetPort)
+	return nil
+}
+
+// UpdateEndpoint creates a K8s endpoint.
+func (d *Deployment) UpdateEndpoint(name, targetIP string, targetPort uint16) error {
+	endpointSpec := &corev1.Endpoints{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: d.namespace,
+		},
+		Subsets: []corev1.EndpointSubset{
+			{
+				Addresses: []corev1.EndpointAddress{
+					{
+						IP: targetIP, // Replace with the desired IP address of the endpoint.
+					},
+				},
+				Ports: []corev1.EndpointPort{
+					{
+						Port: int32(targetPort),
+					},
+				},
+			},
+		},
+	}
+
+	err := d.client.Update(context.Background(), endpointSpec)
+	if err != nil {
+		d.logger.Errorf("error occurred while updating K8s endpoint %s:", err)
+		return err
+	}
+
+	d.logger.Infof("Updating K8s endPoint at %s:%d that connected to external IP: %s:%d", name, targetPort, targetIP, targetPort)
+	return nil
+}
+
+// DeleteEndpoint deletes a k8s endpoint.
+func (d *Deployment) DeleteEndpoint(name string) error {
+	endpointSpec := &corev1.Endpoints{
+		ObjectMeta: metav1.ObjectMeta{Name: name, Namespace: d.namespace}}
+
+	err := d.client.Delete(context.Background(), endpointSpec)
+	if err != nil {
+		d.logger.Errorf("error occurred while deleting K8s endpoint %s:", err)
+		return err
+	}
+
+	d.logger.Infof("Deleting K8s endPoint %s", name)
+	return nil
+}
+
+// GetPodLabelsByIP returns all the labels that match the pod IP.
+func (d *Deployment) GetPodLabelsByIP(podIP string) (map[string]string, error) {
+	var podList corev1.PodList
+	err := d.client.List(context.Background(), &podList) // Adjust the label selector as needed
+	if err != nil {
+		fmt.Printf("GetPodLabelsByIP %s", err.Error())
+		return nil, err
+	}
+
+	for _, pod := range podList.Items {
+		if pod.Status.PodIP == podIP {
+			// Found the pod matching the IP address, so get its labels.
+			return pod.Labels, nil
+		}
+	}
+
+	return nil, fmt.Errorf("pod with IP %s not found", podIP)
+}
+
+// NewDeployment returns a new Kubernetes deployment.
+func NewDeployment() (*Deployment, error) {
+	logger := logrus.WithField("component", "k8s-deployment")
+	cl, err := client.New(config.GetConfigOrDie(), client.Options{})
+	if err != nil {
+		return &Deployment{}, err
+	}
+
+	// Get namespace
```

Comment:
I think this solution is neater:
https://stackoverflow.com/a/57809472

Reply:
Done